src/SFML/Graphics/Text.cpp

```diff
@@ -527,24 +516,31 @@ void Text::ensureGeometryUpdate() const
         // Add the glyph to the vertices
         addGlyphQuad(m_vertices, Vector2f(x, y), m_fillColor, glyph, italicShear);
 
-        // Update the current bounds with the non outlined glyph bounds
-        if (m_outlineThickness == 0)
-        {
-            float left   = glyph.bounds.left;
-            float top    = glyph.bounds.top;
-            float right  = glyph.bounds.left + glyph.bounds.width;
-            float bottom = glyph.bounds.top  + glyph.bounds.height;
-
-            minX = std::min(minX, x + left  - italicShear * bottom);
-            maxX = std::max(maxX, x + right - italicShear * top);
-            minY = std::min(minY, y + top);
-            maxY = std::max(maxY, y + bottom);
-        }
+        // Update the current bounds
+        float left   = glyph.bounds.left;
+        float top    = glyph.bounds.top;
+        float right  = glyph.bounds.left + glyph.bounds.width;
+        float bottom = glyph.bounds.top  + glyph.bounds.height;
+
+        minX = std::min(minX, x + left  - italicShear * bottom);
+        maxX = std::max(maxX, x + right - italicShear * top);
+        minY = std::min(minY, y + top);
+        maxY = std::max(maxY, y + bottom);
 
         // Advance to the next character
         x += glyph.advance + letterSpacing;
     }
 
+    // If we're using outline, update the current bounds
+    if ((m_outlineThickness != 0) && (x > 0))
+    {
+        float outline = std::abs(std::ceil(m_outlineThickness));
```

Comment:
Is `std::abs` necessary here? Do we even support negative outline thicknesses? @eXpl0it3r 
Also, the name `outline` is not very descriptive here. Maybe `absOutlineThickness`?

Reply:
We don't clamp negative thickness, and it's looking weird ( diamond shape, not round ) hence the need for `abs`

Reply:
What about `std::ceil`? Impacted variables are of type `float`, so why not use the absolute thickness without rounding?

Reply:
The documentation states:

> Note that negative values are allowed (so that the outline expands towards the center of the shape)

So yes, we support to an extend negative outline thicknesses

Reply:
The documentation states:

> Be aware that using a negative value for the outline thickness will cause distorted rendering.

So "support" is a strong word, but it's at least a "valid" value, the output is just kind of undefined/not guaranteed

Reply:
`ceil` is  required here because the bounding box of each glyph increases in integer steps, when you increment the outline with fractional values. These are texture sizes we're talking about.